File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.swift
class Solution {
    func numberOfLines(_ widths: [Int], _ s: String) -> [Int] {
        
        var asciiCode = [Int]()
        for i in s {
            if let asciiValue = i.asciiValue {
                asciiCode.append(Int(asciiValue) - 97)
            }
        }
    
        var limit = 0
        var count = 1
        for i in asciiCode {
            limit += widths[i]
            if limit > 100 {
                count += 1
                limit = widths[i]
            }
        }
    
        return [count, limit]

    }
}

File: 824-number-of-lines-to-write-string/number-of-lines-to-write-string.kt
class Solution {
	fun numberOfLines(widths: IntArray, s: String): IntArray {
		val res = IntArray(26)
		var sum = 0
		for (i in s.indices) {
			val width = widths[s[i] - 'a']
			if (sum + width > 100) {
				res[0]++
				sum = width
			} else {
				sum += width
			}
		}
		return intArrayOf(res[0] + 1, sum)
	}
}